src/operation/select-rule.ts

```
@@ -221,20 +223,31 @@ export const selectOptimalRule = (
 
   if (fullSelectableRules.length > 0) {
-    // Select rule with highest filtered count
+    // Select first rule from sorted list (already the best)
     // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-    let bestRule = fullSelectableRules[0]!;
-    for (const rule of fullSelectableRules) {
-      if (rule.filteredCount > bestRule.filteredCount) {
-        bestRule = rule;
-      }
-    }
+    const bestRule = fullSelectableRules[0]!;
```

Biome (1.9.4)

[error] Forbidden non-null assertion.



(lint/style/noNonNullAssertion)